workspace_defs.bzl

```diff
@@ -220,9 +220,9 @@ def google_common_workspace_rules():
     )
 
     _maven_import(
-        artifact = "com.squareup:javapoet:1.11.1",
+        artifact = "com.squareup:javapoet:1.12.1",
         licenses = ["notice"],
-        sha256 = "9cbf2107be499ec6e95afd36b58e3ca122a24166cdd375732e51267d64058e90",
+        sha256 = "e961908e4e3a460ad25acfd240157978b0307acd106691c190efa855cf05eafd",
```

Comment:
I assumed this was supposed to be the sha256 of the [main jar](https://repo1.maven.org/maven2/com/squareup/javapoet/1.12.1/javapoet-1.12.1.jar), but let me know if it should be something else.